radixdlt-engine/src/main/java/com/radixdlt/atom/TxLowLevelBuilder.java

```diff
@@ -271,14 +253,17 @@ public TxLowLevelBuilder read(SubstateId substateId) {
   }
 
   public TxLowLevelBuilder localDown(int index) {
-    var particle = localUpParticles.remove(index);
-    if (particle == null) {
-      throw new IllegalStateException("Local particle does not exist: " + index);
-    }
+    validateParticle(localUpParticles.remove(index), index);
     instruction(REInstruction.REMicroOp.LDOWN, Shorts.toByteArray((short) index));
     return this;
   }
 
+  private void validateParticle(LocalSubstate particle, int index) {
```

Comment:
While we're moving this, can we update the naming? (IE this should be `validateLocalSubstate`)